sierra2mlir/src/libfuncs/mod.rs

```diff
@@ -2150,4 +2159,154 @@ impl<'ctx> Compiler<'ctx> {
 
         Ok(())
     }
+
+    pub fn create_libfunc_null(
+        &'ctx self,
+        func_decl: &LibfuncDeclaration,
+        parent_block: BlockRef<'ctx>,
+        storage: &mut Storage<'ctx>,
+    ) -> Result<()> {
+        let id = func_decl.id.debug_name.as_ref().unwrap().to_string();
+        let arg_type = match &func_decl.long_id.generic_args[0] {
+            GenericArg::UserType(_) => todo!(),
+            GenericArg::Type(type_id) => {
+                storage.types.get(&type_id.id.to_string()).cloned().expect("type to exist")
+            }
+            GenericArg::Value(_) => todo!(),
+            GenericArg::UserFunc(_) => todo!(),
+            GenericArg::Libfunc(_) => todo!(),
```

Comment:
The todo arms should be replaced by a single `_ => unreachable!("Generic arg to null should be a type"),` imo